app/editor/src/features/admin/ingests/configurations/Newspaper.tsx

```diff
@@ -26,6 +27,54 @@ export const Newspaper: React.FC = (props) => {
   const sources = lookups.sources.map((s) => new OptionItem(s.name, s.code, s.isEnabled));
   const source = sources.find((t) => t.value === values.configuration.defaultSource);
 
+  const initialItems = () => {
+    const data: any[] = [];
+    let id = 0;
+    const keyValues = values.configuration.sources.split('&');
+    keyValues.forEach((x: string) => {
+      if (x !== '=') data.push({ id: ++id, name: x.split('=')[0], source: x.split('=')[1] });
+    });
+    data.push({ id: ++id, name: '', source: '' });
+    return data;
+  };
+
+  const [items, setItems] = React.useState<{ id: number; name: string; source: string }[]>(
+    initialItems(),
+  );
+
+  const updateItems = (updatedItems: any[]) => {
+    if (updatedItems.filter((x) => !x.name && !x.source).length === 0)
+      updatedItems.push({ id: updatedItems.length + 1, name: '', source: '' });
+    setItems(updatedItems);
+    setFieldValue(
+      'configuration.sources',
+      updatedItems
+        .filter((t) => !!t.name)
+        .map((item) => `${item.name?.trim()}=${item.source?.trim()}`)
+        .join('&'),
+    );
+  };
+
+  const handleRemove = async (id: number) => {
+    const item = items.find((i) => i.id === id);
+    if (!item) return;
+
+    const updatedItems = items.filter((i) => i.id !== id);
+    updateItems(updatedItems);
+  };
+
+  const onChange = (event: any, cell: any, isSource = true) => {
+    const pointer = event.target.selectionStart;
+    window.requestAnimationFrame(() => {
```

Comment:
What is this pointer and requestAnimationFrame for?

Reply:
Without this there will be an issue when typing in the textbox - when typing a character, the mouse will unexpectedly move to the end of the content.